freqtrade/constants.py

```diff
@@ -94,11 +94,22 @@
         'trailing_stop_positive': {'type': 'number', 'minimum': 0, 'maximum': 1},
         'trailing_stop_positive_offset': {'type': 'number', 'minimum': 0, 'maximum': 1},
         'trailing_only_offset_is_reached': {'type': 'boolean'},
-        'unfilledtimeout': {
+        'unfilled_timeout': {
```

Comment:
This is higly problematic. You're breaking everyones strategy / configuration.
While i agree, `unfilled_timeout` looks better - i don't want to have everyone modify configuration and/or strategy for no benefit.


> Please note that this PR, as it stands introduces no change for most users since it defaults to existing behavior (cancels buy orders based on unfilled_timeout).

This is not true - as you've made my configuration i had for "unfilledtimeout" invalid.
Since the new configuration has defaults, users won't even notice (maybe not for months) - until the order uses the default timeout while the user expected it not to because it was changed to "something else" (higher or lower).